Source/Avionics/ControlLocker.cs

```diff
@@ -13,28 +11,25 @@ class ControlLockerUtils
     {
         public static bool ShouldLock(List<Part> parts, bool countClamps, out float maxMass, out float vesselMass)
         {
-            int crewCount = -1;
-            maxMass = vesselMass = 0f;
-            Part p;
+            maxMass = vesselMass = 0f;  // These are unreliable if the calculation exits early!
 
             if (parts == null || parts.Count <= 0)
                 return false;
+            if (!HighLogic.LoadedSceneIsFlight && !HighLogic.LoadedSceneIsEditor) return false;
 
-            for (int i = parts.Count - 1; i >= 0; --i)
+            int crewCount = (HighLogic.LoadedSceneIsFlight) ? parts[0].vessel.GetCrewCount() : CrewAssignmentDialog.Instance.GetManifest().GetAllCrew(false).Count;
+
+            foreach (Part p in parts)
```

Comment:
Umm, I think there was a reason (performance related, probably garbage generation) why `foreach` was avoided here.

Reply:
Foreach is safe for List<> types for both GC and execution time.  Also this implementation has previously been capped at one execution/second.

Reply:
OK, I just wanted to be sure it had been considered :)

Reply:
For reference: https://jacksondunstan.com/articles/3805